lib/src/main/java/graphql/nadel/Nadel.kt

```diff
@@ -226,75 +218,97 @@ class Nadel private constructor(
-            this.instrumentation = instrumentation
+        fun underlyingSchema(serviceName: String, nsdl: Reader): Builder {
+            schemaBuilder.underlyingSchema(serviceName, nsdl)
             return this
         }
 
-        fun preparsedDocumentProvider(preparsedDocumentProvider: PreparsedDocumentProvider): Builder {
-            this.preparsedDocumentProvider = preparsedDocumentProvider
+        fun underlyingSchema(serviceName: String, nsdl: String): Builder {
+            schemaBuilder.underlyingSchema(serviceName, nsdl)
             return this
         }
 
-        fun executionIdProvider(executionIdProvider: ExecutionIdProvider): Builder {
-            this.executionIdProvider = executionIdProvider
+        @JvmName("underlyingSchemasReader")
+        fun underlyingSchemas(underlyingSchemas: Map<String, Reader>): Builder {
+            schemaBuilder.underlyingSchemas(underlyingSchemas)
             return this
         }
 
-        fun serviceExecutionHooks(serviceExecutionHooks: ServiceExecutionHooks): Builder {
-            this.serviceExecutionHooks = serviceExecutionHooks
+        @JvmName("underlyingSchemasString")
+        fun underlyingSchemas(underlyingSchemas: Map<String, String>): Builder {
+            schemaBuilder.underlyingSchemas(underlyingSchemas)
             return this
         }
 
         fun overallWiringFactory(wiringFactory: WiringFactory): Builder {
-            overallWiringFactory = wiringFactory
+            schemaBuilder.overallWiringFactory(wiringFactory)
             return this
         }
 
         fun underlyingWiringFactory(wiringFactory: WiringFactory): Builder {
-            underlyingWiringFactory = wiringFactory
+            schemaBuilder.underlyingWiringFactory(wiringFactory)
             return this
         }
 
         fun schemaTransformationHook(hook: SchemaTransformationHook): Builder {
-            schemaTransformationHook = hook
+            schemaBuilder.schemaTransformationHook(hook)
+            return this
+        }
+
+        fun serviceExecutionFactory(serviceExecutionFactory: ServiceExecutionFactory): Builder {
+            this.serviceExecutionFactory = serviceExecutionFactory
+            schemaBuilder.serviceExecutionFactory(serviceExecutionFactory)
+            return this
+        }
+
+        fun instrumentation(instrumentation: NadelInstrumentation): Builder {
+            this.instrumentation = instrumentation
+            return this
+        }
+
+        fun preparsedDocumentProvider(preparsedDocumentProvider: PreparsedDocumentProvider): Builder {
+            this.preparsedDocumentProvider = preparsedDocumentProvider
+            return this
+        }
+
+        fun executionIdProvider(executionIdProvider: ExecutionIdProvider): Builder {
+            this.executionIdProvider = executionIdProvider
+            return this
+        }
+
+        fun serviceExecutionHooks(serviceExecutionHooks: ServiceExecutionHooks): Builder {
```

Comment:
Just moved the members around so all the schema builder stuff is together. It's the same API wise.